src/tasks/submit/flashbots.rs

```diff
@@ -117,38 +117,56 @@ impl FlashbotsTask {
                 debug!("upstream task gone - exiting flashbots task");
                 break;
             };
-            let span = sim_result.span();
-            span_debug!(span, "received sim result");
+
+            let span = sim_result.sim_env.clone_span();
+
+            // Don't submit empty blocks
+            if sim_result.block.is_empty() {
+                counter!("signet.builder.flashbots.empty_block").increment(1);
+                span_debug!(span, "received empty block - skipping");
+                continue;
+            }
+            span_debug!(span, "flashbots task received block");
 
             // Prepare a MEV bundle with the configured call type from the sim result
-            let Ok(bundle) =
+            let result =
                 self.prepare(&sim_result).instrument(span.clone()).await.inspect_err(|error| {
                     counter!("signet.builder.flashbots.bundle_prep_failures").increment(1);
                     span_debug!(span, %error, "bundle preparation failed");
-                })
-            else {
-                continue;
+                });
+            let bundle = match result {
+                Ok(bundle) => bundle,
+                Err(_) => continue,
```

Comment:
this could be further simplified with:

```
let bundle = match self.prepare(&sim_result)
    .instrument(span.clone())
    .await
    .inspect_err(|error| {
        counter!("signet.builder.flashbots.bundle_prep_failures").increment(1);
        span_debug!(span, %error, "bundle preparation failed");
    }) {
    Ok(bundle) => bundle,
    Err(_) => continue,
};
```